File: algo/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: 'List[str]') -> 'List[List[str]]':
        helper_dict = {}
        for word in strs:
            tempStr = ''.join(sorted(word))
            if helper_dict.get(tempStr) is None:
                helper_dict[tempStr] = [word]
            else:
                helper_dict[tempStr].append(word)
        return list(helper_dict.values())

File: round2/49. Group Anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var res = new List<IList<string>>();
        if (strs.Length == 0) return res;
        var helperDict = new Dictionary<string, List<string>>();
        foreach (var str in strs) {
            string orderedStr = new string (str.OrderBy(c => c).ToArray());
            if (!helperDict.ContainsKey(orderedStr)) {
                helperDict[orderedStr] = new List<string>();
            }
            helperDict[orderedStr].Add(str);
        }
        foreach(KeyValuePair<string, List<string>> entry in helperDict)
        {
            res.Add(entry.Value);
        }
        return res;
    }
}